src/pages/guildverse.tsx

```diff
@@ -0,0 +1,106 @@
+import { SimpleGrid } from "@chakra-ui/react"
+import BrainCard from "components/brain/BrainCard"
+import FilterSelect, { FilterOption } from "components/brain/FilterSelect"
+import Layout from "components/common/Layout"
+import LinkPreviewHead from "components/common/LinkPreviewHead"
+import CategorySection from "components/explorer/CategorySection"
+import SearchBar from "components/explorer/SearchBar"
+import { GetServerSideProps } from "next"
+import { useMemo, useState } from "react"
+import { BrainCardData } from "types"
+
+type Props = {
+  cards: BrainCardData[]
+}
+
+export const FILTER_OPTIONS: Array<FilterOption> = [
+  { value: "requirement", label: "requirement" },
+  { value: "reward", label: "reward" },
+  { value: "built with Guild", label: "built with Guild" },
+  { value: "web2", label: "web2" },
+  { value: "web3", label: "web3" },
+]
+
+const Guildverse = ({ cards: cards }: Props): JSX.Element => {
+  const [search, setSearch] = useState<string>("")
+  const [filterData, setFilterData] = useState<Array<FilterOption>>([])
+  const renderedCards = useMemo(
+    () =>
+      cards.filter(
+        (card) =>
+          filterData.every((filterTag) => card?.tags.includes(filterTag.value)) &&
+          card.title.toLowerCase().includes(search.toLowerCase())
+      ),
+    [cards, filterData, search]
+  )
+
+  return (
+    <>
+      <LinkPreviewHead path="" />
+      <Layout title="Guildverse">
+        <SimpleGrid
+          templateColumns={{ md: "2fr 3fr" }}
+          gap={{ base: 2, md: "6" }}
+          mb={8}
+        >
+          <SearchBar placeholder="Search" {...{ search, setSearch }} />
+          <FilterSelect {...{ FILTER_OPTIONS, setFilterData }} />
```

Comment:
No need to spread `FILTER_OPTIONS` here. And then you could pass the  `setFilterData` the way you'd pass any other prop:

```tsx
<FilterSelect setFilterData={setFilterData} />
```